aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1048 @@
+        elgamal_pub_key_bulk: vector<vector<vector<u8>>>,
+        network_port_bulk: vector<vector<u16>>,
+        rpc_por_bulkt: vector<vector<u16>>,
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Assert the length of the vector for two are the same
+        let ids_len = vector::length(&ids);
+        assert!(
+            ids_len == vector::length(&node_addresses_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&ip_public_address_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&dora_public_key_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&network_public_key_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&elgamal_pub_key_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&network_port_bulk),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            ids_len == vector::length(&rpc_por_bulkt),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        while (vector::length(&ids) > 0) {
+            let id = vector::pop_back(&mut ids);
+            let node_addresses = vector::pop_back(&mut node_addresses_bulk);
+            let ip_public_address = vector::pop_back(&mut ip_public_address_bulk);
+            let dora_public_key = vector::pop_back(&mut dora_public_key_bulk);
+            let network_public_key = vector::pop_back(&mut network_public_key_bulk);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key_bulk);
+            let network_port = vector::pop_back(&mut network_port_bulk);
+            let rpc_port = vector::pop_back(&mut rpc_por_bulkt);
+            upsert_committee(
+                com_store_addr,
+                owner_signer,
+                id,
+                node_addresses,
+                ip_public_address,
+                dora_public_key,
+                network_public_key,
+                elgamal_pub_key,
+                network_port,
+                rpc_port
+            );
+        }
+    }
+
+    /// Remove the committee from the store
+    public entry fun remove_committee(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let (id, committee_info) = simple_map::remove(&mut committee_store.committee_map, &id);
```

Comment:
Does this `remove` operation ever result in failure? If so, it should be handled via `assert` first.

Reply:
Right, added